packages/config-cli/src/cli/bmc/import.config.ts

```diff
@@ -0,0 +1,120 @@
+import { Env, fsa, LogConfig } from '@basemaps/shared';
+import { ConfigBundled, ConfigImagery, ConfigProvider, ConfigTileSet, ConfigVectorStyle } from '@basemaps/config';
+import { CommandLineAction, CommandLineFlagParameter, CommandLineStringParameter } from '@rushstack/ts-command-line';
+import { Q, Updater } from './config/base.config.js';
+import { StyleUpdater } from './config/updater/style.updater.js';
+import { ProviderUpdater } from './config/updater/provider.updater.js';
+import { TileSetUpdater } from './config/updater/tileset.updater.js';
+import { invalidateCache } from '@basemaps/cli/build/cli/util.js';
+import { ImageryUpdater } from './config/updater/imagery.updater.js';
+import { ImageryTileSetUpdater } from './config/updater/imagery.tileset.updater.js';
+
+export enum UpdaterType {
+  Style = 'style',
+  TileSet = 'tileset',
+  Imagery = 'imagery',
+  ImageryTileSet = 'imagery_tileset',
+  Provider = 'provider',
+}
+
+export class CommandImport extends CommandLineAction {
+  private config: CommandLineStringParameter;
+  private commit: CommandLineFlagParameter;
+
+  promises: Promise<boolean>[] = [];
+  /** List of paths to invalidate at the end of the request */
+  invalidations: string[] = [];
+
+  public constructor() {
+    super({
+      actionName: 'import',
+      summary: 'import a config json into dynamodb',
+      documentation: 'Given a valid bundle config json and import them into dynamodb',
+    });
+  }
+
+  protected onDefineParameters(): void {
+    this.config = this.defineStringParameter({
+      argumentName: 'CONFIG',
+      parameterLongName: '--config',
+      description: 'Path of config json',
+      required: true,
+    });
+    this.commit = this.defineFlagParameter({
+      parameterLongName: '--commit',
+      description: 'Actually start the import',
+      required: false,
+    });
+  }
+
+  async onExecute(): Promise<void> {
+    const logger = LogConfig.get();
+    logger.level = 'trace';
+    const commit = this.commit.value ?? false;
+    const config = this.config.value;
+    if (config == null) throw new Error('Please provide a config json');
+
+    const HostPrefix = Env.isProduction() ? '' : 'dev.';
+    const healthEndpoint = `https://${HostPrefix}basemaps.linz.govt.nz/v1/health`;
+
+    logger.info({ url: healthEndpoint }, 'Import:ValidateHealth');
+    if (commit) {
+      const res = await fetch(healthEndpoint);
+      if (!res.ok) throw new Error('Cannot update basemaps is unhealthy');
+    }
+
+    logger.info({ config: config }, 'Import:Load');
+    const configJson = await fsa.readJson<ConfigBundled>(config);
+
+    logger.info({ config: config }, 'Import:Style');
+    for await (const config of configJson.style) this.update(UpdaterType.Style, config, commit);
```

Comment:
I think we can greatly simplify this now.

As this configuration has already been "imported" it has already been validated and the record can basically just be inserted.

I don't think we need a "updater" per type, we just need to query the database for the id, then diff and then insert by id.